File: C++/Guess the number.cpp
#include <iostream>
#include <cstdlib>
#include <ctime>
using namespace std;

int main() {
    srand(time(NULL)); // Seed the random number generator

    int secretNumber = rand() % 100 + 1; // Generate a random number between 1 and 100
    int guess;
    int numTries = 0;
    bool hasWon = false;

    cout << "Welcome to the Guessing Game!" << endl;
    cout << "You have 7 attempts to guess the secret number between 1 and 100." << endl;

    while (numTries < 7) {
        cout << "Enter your guess: ";
        cin >> guess;
        numTries++;

        if (guess == secretNumber) {
            hasWon = true;
            break;
        } else if (guess < secretNumber) {
            cout << "Too low! Guess again." << endl;
        } else {
            cout << "Too high! Guess again." << endl;
        }
    }

    if (hasWon) {
        cout << "Congratulations! You guessed the secret number in " << numTries << " tries." << endl;
    } else {
        cout << "Sorry, you didn't guess the secret number. The secret number was " << secretNumber << "." << endl;
    }

    return 0;
}

File: JavaScript/guess the number.js
<!DOCTYPE html>
<html>
<head>
  <title>Guess the Number Game</title>
</head>
<body>
  <h1>Guess the Number Game</h1>
  <p>Guess a number between 1 and 100:</p>
  <input type="text" id="guess" />
  <button onclick="checkGuess()">Guess</button>
  <p id="result"></p>
  <script>
    // generate a random number between 1 and 100
    const randomNumber = Math.floor(Math.random() * 100) + 1;

    function checkGuess() {
      const guess = parseInt(document.getElementById("guess").value);

      if (isNaN(guess) || guess < 1 || guess > 100) {
        document.getElementById("result").textContent = "Invalid guess. Please enter a number between 1 and 100.";
        return;
      }

      if (guess === randomNumber) {
        document.getElementById("result").textContent = "Congratulations! You guessed the correct number.";
      } else if (guess < randomNumber) {
        document.getElementById("result").textContent = "Your guess is too low. Try again.";
      } else {
        document.getElementById("result").textContent = "Your guess is too high. Try again.";
      }
    }
  </script>
</body>
</html>
